src/main/webapp/scripts/classes/comment-class.js

```diff
@@ -0,0 +1,32 @@
+/*
+ *  Class handles actions related to creating and analyzing new comments
+ *  TODO: add methods to add new comment to youtube video using Youtube API
+ */
+
+class Comment{
+  #MAX_COMMENTS  = 3000;
+  #NO_COMMENT_ERROR = 'No comment inputted';
+  #COMMENT_LENGTH_EXCEEDED_ERROR = 'Comment is too long. We are currently only able to analyze comments of up to approx. 3000 characters.';
+  constructor(text) {
+    this.text = text;
+    this.isValid = !this.getError();
+    this.perspectiveString = this.makePerspectiveRequestString();
+  }
+  getError() {
+    if(this.text == 0) {
+      return this.#NO_COMMENT_ERROR;
+    }
+    if(this.text.length > this.#MAX_COMMENTS) {
+      return this.#COMMENT_LENGTH_EXCEEDED_ERROR;
```

Comment:
Returning strings for errors seems odd. How about throw JS errors? Or not even letting the constructor finish creating the comment object and throwing an error telling the caller that they called it incorrectly?

Reply:
I changed it to throw errors and the code looks a lot better, thanks for the suggestion! I did not think of making my own error objects